.github/workflows/rust.yml

```diff
@@ -45,6 +45,30 @@ jobs:
           command: check
           args: --all-targets --no-default-features --workspace
 
+  wasm:
+    name: Check WASM compatibility
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          override: true
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v1.3.0
+
+      - name: Check all features
+        uses: actions-rs/cargo@v1.0.3
+        with:
+          command: check
+          args: --all-targets --all-features --target wasm32-unknown-unknown
```

Comment:
this looks funny but `--all-targets` actually imply that is compiles for `test` and `bench` I guess?

Reply:
Yeah that's my understanding of it! It's not really important to have here but I thought we may as well try and push for _everything_ to compile to wasm when possible.. _shrug_ :)